android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/CloudRepository.java

```diff
@@ -358,8 +358,9 @@ protected JSONObject wrapKmpKeyboardJSON(JSONArray languagesArray) {
 
   protected List<Keyboard> processKeyboardJSON(JSONObject query, boolean fromKMP) {
     List<Keyboard> keyboardsList = new ArrayList<>();
-    //keyboardModifiedDates = new HashMap<String, String>();
-
+    if (query.length() == 0) {
```

Comment:
This guards when query is `{}` and was causing `query.getJSONObject()` in line 368 to throw